python/lib/sift_py/data_import/_config.py

```diff
@@ -71,7 +73,10 @@ def convert_format(cls, raw: Union[str, TimeFormatType]) -> str:
             if value is not None:
                 return value.as_human_str()
 
-        raise PydanticCustomError("invalid_config_error", f"Invalid time format: {raw}.")
+        raise PydanticCustomError(
+            cast(LiteralString, "invalid_config_error"),
```

Comment:
Just curious why is pyright complaining about this? This seems kind of aggressive.

Reply:
Oops sorry I missed this comment earlier -- I ended up reading/skimming a lot of discourse about this, and it seems like it's really to do with the fact that there's a lot of fundamental disagreement on type handling in python, and `LiteralString` seems to a fairly new construct that there isn't yet much agreement on how to handle, so pyright seems to have picked one direction. It seems strange to me as well. I'm a little meh on the cast, so we could silence pyright or just look into options for making it less strict in this case. Any strong opinions on the path here?

Reply:
My vote would be to silence these.

Reply:
Ah I probably should have read this earlier: https://peps.python.org/pep-0675/
It looks like the intent is to make the inputs be `Literal["foo"]`, so I think I just did this in a super subpar way. Otherwise I don't disagree, I think for our use case an ignore is fine. I looks like we should be able to go the `Literal["foo"]` route as well if we want as that seems closer to the intention with this type. Let me see what pyright thinks of that.

Reply:
If it turns out it's pyright just being altogether too strict then I will also vote for just silencing.

Reply:
I added more details of what I've understood to our slack thread. For now I just pushed the version up that suppresses these, but let's chat more in slack if you want to get into the details more!